src/shell/lean.cpp

```diff
@@ -289,6 +298,32 @@ options set_config_option(options const & opts, char const * in) {
     }
 }
 
+void load_plugin(std::string path) {
+    void * init;
+    // we never want to look up plugins using the system library search
+    path = lrealpath(path);
+#ifdef LEAN_WINDOWS
+    HMODULE h = LoadLibrary(path.c_str());
+    if (!h) {
+        throw exception(sstream() << "error loading plugin " << path);
+    }
+    init = reinterpret_cast<void *>(GetProcAddress(h, "initialize_Default"));
+#else
+    void *handle = dlopen(path.c_str(), RTLD_LAZY);
+    if (!handle) {
+        throw exception(sstream() << "error loading plugin, " << dlerror());
+    }
+    init = dlsym(handle, "initialize_Default");
```

Comment:
It seems every plugin must define `initialize_Default`. Is this a problem?
https://stackoverflow.com/questions/6538501/linking-two-shared-libraries-with-some-of-the-same-symbols
I am assuming it should work since you are using `dlsym`.

Reply:
Yes, objects loaded by `dlsym` are independent by default.
```
      RTLD_LOCAL
              This is the converse of RTLD_GLOBAL, and the default if neither flag is specified.  Symbols  defined  in  this
              shared object are not made available to resolve references in subsequently loaded shared objects.
```
As far as I could find out, this is actually the *only* behavior supported on Windows, so we should be fine.

Btw, note that if we ever do prefix import paths with the package name as we once discussed, this would be `initialize_SnakeLinter_Default` anyway (which `--plugin` should derive from the library filename).

Reply:
Should this try calling dlopen with RTLD_NOLOAD (if available) first, to check if the shared library is actually loaded already,
to avoid opening the same library twice?

Reply:
`If the same shared object is opened again with dlopen(), the same object handle  is  returned`. And Lean initializers are idempotent.

Reply:
Thanks I had missed the section describing that in the docs, needed to read further.